lib/bolt/transport/ssh/connection.rb

```diff
@@ -246,6 +268,19 @@ def net_ssh_verifier(verifier)
           end
         end
 
+        # Add all default algorithms if the 'defaults' key is present and filter
+        # out any unsupported algorithms.
+        def net_ssh_algorithms(type, algorithms)
+          if algorithms.include?('defaults')
+            defaults = Net::SSH::Transport::Algorithms::DEFAULT_ALGORITHMS[type]
+            algorithms += defaults
+          end
+
+          known = Net::SSH::Transport::Algorithms::ALGORITHMS[type]
+
+          algorithms & known
```

Comment:
Until #1861 is in and released, it'd be nice if we warned here that certain specified algorithms aren't supported at all.
```
supported, unsupported = algorithms.partition { |a| known.include?(a) }
@logger.warn("These #{type.to_s} algorithms aren't supported by the net-ssh library: #{unsupported}")
supported
```

Reply:
My only issue with warning about unsupported algorithms is that it will warn for every single target. If we want #1861 in as an alternate to warning, then I think we can just block this ticket until that's ready.

Reply:
Added the work for #1861 to this PR. Felt the error message should also list disabled algorithms and how to enable them, so it's closely tied to the work for #1862 .